plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformUtils.kt

```diff
@@ -214,15 +213,19 @@ suspend fun JobId.pollTransformationStatusAndPlan(
                 }
                 state = newStatus
                 numRefreshes = 0
+                return@waitUntil state
             } catch (e: AccessDeniedException) {
                 if (numRefreshes++ > maxRefreshes) throw e
                 refreshToken(project)
+                return@waitUntil state
             } finally {
                 sleep(sleepDurationMillis)
             }
         }
     } catch (e: WaiterUnrecoverableException) {
```

Comment:
Is there a reason we can't just catch any `Exception` at once here?

Reply:
Changed to generic Exception so that we can handle all exceptions thrown.

Also made the fix so that the status will update when job failed due to exceptions.